From 7396b7eb907bf0d22197e0f1e127f25e4af1c7b5 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Fri, 07 Feb 2020 16:51:13 +0200
Subject: [PATCH] [SW-8229] goya2: split linker for each target

To better support onix which cant load rom we split
the linker script for each target. Now onix will load
the arcboot_rom to ScratchPad SRAM

Change-Id: Icac829bac02fe0d0653b47520f840de262b34c42
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/Makefile b/Makefile
index fe964fe..7729326 100644
--- a/Makefile
+++ b/Makefile
@@ -27,8 +27,8 @@
 ARCBOOT_CFLAGS += -DBTLID=0x30
 
 LDFLAGS = -flto
-LDFLAGS += -L $(LIBDIR) -lsb -lellipsys
-LDFLAGS += -T $(HL_DEV)/$@.lds
+LDFLAGS += -L $(LIBDIR) -lsb -lellipsys -L $(HL_DEV)
+LDFLAGS += -T $(HL_DEV)/$(HL_TARGET)_$@.lds
 LDFLAGS += -nostartfiles
 LDFLAGS += -Wl,-e__start
 
diff --git a/goya2_arc/asic_arcboot_ram.lds b/goya2_arc/asic_arcboot_ram.lds
new file mode 100644
index 0000000..c960869
--- /dev/null
+++ b/goya2_arc/asic_arcboot_ram.lds
@@ -0,0 +1,8 @@
+/*
+ * (C) Copyright 2017 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+ INCLUDE arcboot_ram.lds
\ No newline at end of file
diff --git a/goya2_arc/asic_arcboot_rom.lds b/goya2_arc/asic_arcboot_rom.lds
new file mode 100644
index 0000000..d49acc6
--- /dev/null
+++ b/goya2_arc/asic_arcboot_rom.lds
@@ -0,0 +1,8 @@
+/*
+ * (C) Copyright 2017 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+ INCLUDE arcboot_rom.lds
\ No newline at end of file
diff --git a/goya2_arc/onix_arcboot_ram.lds b/goya2_arc/onix_arcboot_ram.lds
new file mode 100644
index 0000000..c960869
--- /dev/null
+++ b/goya2_arc/onix_arcboot_ram.lds
@@ -0,0 +1,8 @@
+/*
+ * (C) Copyright 2017 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+ INCLUDE arcboot_ram.lds
\ No newline at end of file
diff --git a/goya2_arc/onix_arcboot_rom.lds b/goya2_arc/onix_arcboot_rom.lds
new file mode 100644
index 0000000..908c04b
--- /dev/null
+++ b/goya2_arc/onix_arcboot_rom.lds
@@ -0,0 +1,13 @@
+/*
+ * (C) Copyright 2017 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+INCLUDE arcboot_rom.lds
+
+MEMORY
+{
+	SBROM	: ORIGIN = 0x1BFE1000, LENGTH = 60K
+}
\ No newline at end of file
diff --git a/goya2_arc/pldm_arcboot_ram.lds b/goya2_arc/pldm_arcboot_ram.lds
new file mode 100644
index 0000000..c960869
--- /dev/null
+++ b/goya2_arc/pldm_arcboot_ram.lds
@@ -0,0 +1,8 @@
+/*
+ * (C) Copyright 2017 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+ INCLUDE arcboot_ram.lds
\ No newline at end of file
diff --git a/goya2_arc/pldm_arcboot_rom.lds b/goya2_arc/pldm_arcboot_rom.lds
new file mode 100644
index 0000000..d49acc6
--- /dev/null
+++ b/goya2_arc/pldm_arcboot_rom.lds
@@ -0,0 +1,8 @@
+/*
+ * (C) Copyright 2017 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+ INCLUDE arcboot_rom.lds
\ No newline at end of file
